sdk/identity/identity/src/credentials/managedIdentityCredential/tokenExchangeMsi.ts

```diff
@@ -105,12 +117,13 @@ export function tokenExchangeMsi(): MSI {
         );
       }
 
-      return msiGenericGetToken(
-        identityClient,
-        prepareRequestOptions(scopes, assertion, clientId || process.env.AZURE_CLIENT_ID),
-        expiresInParser,
-        getTokenOptions
-      );
+      const request = createPipelineRequest({
+        abortSignal: getTokenOptions.abortSignal,
+        ...prepareRequestOptions(scopes, assertion, clientId || process.env.AZURE_CLIENT_ID!),
+        allowInsecureConnection: true
```

Comment:
Same comment about documenting why this is necessary.

Reply:
IMDS endpoints are http, which are blocked by default by our pipelines.